daft/dataframe/display.py

```diff
@@ -0,0 +1,55 @@
+from daft.context import get_context
+
+
+class MermaidFormatter:
+    def __init__(self, builder, show_all: bool = False, simple: bool = False, is_cached: bool = False):
+        self.builder = builder
+        self.show_all = show_all
+        self.simple = simple
+        self.is_cached = is_cached
+
+    def _repr_markdown_(self):
+        builder = self.builder
+        output = ""
+        display_opts = {
+            "simple": self.simple,
+            "format": "mermaid",
+        }
+
+        # TODO handle cached plans
```

Comment:
is this something we still have to handle?

Reply:
yeah I'd like to add cached plans like we do with the normal display